source/upgrade.txt

```diff
@@ -0,0 +1,95 @@
+
+Overview
+--------
+
+This page describes the changes you must make to your application
+when you upgrade to a new version of the {+driver-short+}.
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new {+driver-short+} version is compatible with the {+mdb-server+} versions
+  your application connects to and the C++ standard version your
+  application compiles with. For version compatibility information, see the
+  :ref:`{+driver-short+} Compatibility <cpp-compatibility-tables>`
+  page.
+- Address any breaking changes between the driver version
+  your application is using and your planned upgrade version in the
+  :ref:`Breaking Changes <cpp-breaking-changes>` section.
+
+.. tip::
+
+   To ensure compatibility across {+mdb-server+} versions when
+   upgrading driver versions, use the
+   :ref:`{+stable-api+} <cpp-stable-api>`.
+
+.. _cpp-breaking-changes:
+
+Breaking Changes
+----------------
+
+.. note::
+
+   The {+driver-short+} follows :wikipedia:`semantic versioning. <Software_versioning#Schemes>`
+   The driver might break build system or package compatibility in any release, and
+   patch releases can revert accidental API breaking changes.
+
+   For more information, see :ref:`<cpp-api-abi-versioning>`.
+
+A breaking change is a change of a convention or a behavior starting in a specific
+version of the driver. This type of change may prevent your application from working
+properly if not addressed before upgrading the driver.
+
+The breaking changes in this section are categorized by the driver version that introduced
+them. When upgrading driver versions, address all the breaking changes between the current
+and upgrade versions.
+
+.. example:: Upgrading from Version 3.0
+
+  If you're upgrading the {+driver-short+} from version 3.0 to version 3.10,
+  address all breaking changes listed for versions 3.1 to 3.10, if any.
+
+.. _version-3.10-breaking-changes:
+
+Version 3.10 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This driver version introduces the following breaking changes:
+
+- Drops support for the ``find_package(libbsoncxx)`` and ``find_package(libmongocxx)`` CMake
+  commands. Call ``find_package(bsoncxx)`` and ``find_package(mongocxx)``
+  instead.
+- Drops support for the ``LIBBSONCXX_*`` and ``LIBMONGOCXX_*`` CMake variables
+  provided by the legacy CMake package config files. Use the ``mongo::bsoncxx_*`` and
+  ``mongo::mongocxx_*`` CMake targets instead.
+- Removes the experimental C++ standard library (``BSONCXX_POLY_USE_STD_EXPERIMENTAL``) as a
+  polyfill option.
+
+.. _version-3.9-breaking-changes:
+
+Version 3.9 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This driver version introduces the following breaking changes:
+
+- Removes support for exported targets from the CMake project build tree.
+- Drops support for macOS 10.14, macOS 10.15, and Ubuntu 14.04.
+- Requires MongoDB C Driver v1.25.0 or later.
+- Requires CMake v3.15 or later
```

Comment:
Rather than committing to a definitive list of breaking changes for 3.8 and earlier or omitting their mention entirely, recommend adding a section like "Version 3.8 and Earlier" that defers to the old release notes on GitHub, e.g.

> ## Version 3.8 and Earlier
>
> For driver versions 3.8 and earlier, see the release notes and associated JIRA tickets for each release: https://github.com/mongodb/mongo-cxx-driver/releases

This way readers will have somewhere to look and be redirected rather than hitting a dead-end.

Reply:
Good idea.